packages/perseus/src/components/tooltip.tsx

```diff
@@ -40,136 +41,135 @@
 // TODO(joel/aria) fix z-index issues https://s3.amazonaws.com/uploads.hipchat.com/6574/29028/yOApjwmgiMhEZYJ/Screen%20Shot%202014-05-30%20at%203.34.18%20PM.png
 // z-index: 3 on perseus-formats-tooltip seemed to work
 
-import createReactClass from "create-react-class";
-import PropTypes from "prop-types";
 import * as React from "react";
 import ReactDOM from "react-dom";
 
-const zIndex = 10;
-
-const Triangle = createReactClass({
-    propTypes: {
-        color: PropTypes.string.isRequired,
-        left: PropTypes.number.isRequired,
-        top: PropTypes.number.isRequired,
-        width: PropTypes.number.isRequired,
-        height: PropTypes.number.isRequired,
-        horizontalDirection: PropTypes.oneOf(["left", "right"]).isRequired,
-        verticalDirection: PropTypes.oneOf(["top", "bottom"]).isRequired,
-    },
-
-    render: function () {
-        let borderLeft;
-        let borderRight;
-        let borderTop;
-        let borderBottom;
-
-        const hBorder = `${this.props.width}px solid transparent`;
-        if (this.props.horizontalDirection === "right") {
-            borderLeft = hBorder;
-        } else {
-            borderRight = hBorder;
-        }
-
-        const vBorder = `${this.props.height}px solid ${this.props.color}`;
-        if (this.props.verticalDirection === "top") {
-            borderTop = vBorder;
-        } else {
-            borderBottom = vBorder;
-        }
-
-        return (
-            <div
-                style={{
-                    display: "block",
-                    height: 0,
-                    width: 0,
-                    position: "absolute",
-                    left: this.props.left,
-                    top: this.props["top"],
-                    borderLeft: borderLeft,
-                    borderRight: borderRight,
-                    borderTop: borderTop,
-                    borderBottom: borderBottom,
-                }}
-            />
-        );
-    },
-});
-
-const TooltipArrow = createReactClass({
-    propTypes: {
-        position: PropTypes.string,
-        visibility: PropTypes.string,
-        left: PropTypes.number,
-        top: PropTypes.number,
-        color: PropTypes.string.isRequired, // a css color
-        border: PropTypes.string.isRequired, // a css color
-        width: PropTypes.number.isRequired,
-        height: PropTypes.number.isRequired,
-        horizontalDirection: PropTypes.oneOf(["left", "right"]).isRequired,
-        verticalDirection: PropTypes.oneOf(["top", "bottom"]).isRequired,
-    },
+import type {Property} from "csstype";
 
-    getDefaultProps: function () {
-        return {
-            position: "relative",
-            visibility: "visible",
-            left: 0,
-            top: 0,
-        };
-    },
+const zIndex = 10;
 
+type HorizontalDirection = "left" | "right";
+type VerticalDirection = "top" | "bottom";
```

Comment:
I tried, but it gets complicated because we use these values in the config objects (`VERTICAL_CORNERS`, `HORIZONTAL_CORNERS`, `HORIZONTAL_ALIGNMNENTS`). I'm going to leave it like this for now.

Reply:
Why does having these values in config objects complicate things?